core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -578,6 +578,10 @@ public TableMetadata removeSnapshotsIf(Predicate<Snapshot> removeIf) {
     return new Builder(this).removeSnapshots(toRemove).build();
   }
 
+  public TableMetadata updateMetadataLocation(String metadataFileLocation) {
```

Comment:
I decided to add this functionality to `TableMetadata` because we would really only want to update the metadata location in this case and not do any other updates that would require updating `lastUpdatedMillis`. 
An alternative would be to have 

```
public static Builder buildFrom(TableMetadata base, String metadataFileLocation) {
    return new Builder(base, metadataFileLocation);
  }
```

but that would allow other method calls on the builder, which in turn would also require updating `lastUpdatedMillis`.